src/Microsoft.OData.Core/ODataLibraryCompatibility.cs

```diff
@@ -39,14 +39,20 @@ public enum ODataLibraryCompatibility
         /// </summary>
         DoNotThrowExceptionForTopLevelNullProperty = 1 << 3,
 
+        /// <summary>
+        /// When enabled, OData inner error is serialized with "message", "type", and "stacktrace" properties irrespective of whether
+        /// they are null or empty. Also, nested inner error is serialized with property name as "internalexception".
+        /// </summary>
+        UseLegacyODataInnerErrorSerialization = 1 << 4,

```

Comment:
Do we have a test for this new flag?

Reply:
There's a test here: https://github.com/OData/odata.net/pull/2957/files#diff-afe9d5bc87d1806f378774f576fae48802d45003546b8c789fa6344bd554c001R96

Reply:
@ElizabethOkerio Yes. I went to existing tests and parameterized them such that they are executed both when the flag is toggled and when it's not